src/protocol/command.rs

```diff
@@ -28,8 +44,32 @@ impl Command {
         word.chars().map(escape_char).collect::<Vec<_>>().join("")
     }
 
+    fn executable_with_path(executable: &[u8]) -> PathBuf {
```

Comment:
Can we rename this to something with `canonicalize` in the name?

Reply:
As mentioned in my other comment, I renamed this function to `canonicalize_executable_path`.